torchgeo/datasets/idtrees.py

```diff
@@ -0,0 +1,532 @@
+        "QUHE2": "Quercus hemisphaerica W. Bartram ex Willd.",
+        "QULA2": "Quercus laevis Walter",
+        "QULA3": "Quercus laurifolia Michx.",
+        "QUMO4": "Quercus montana Willd.",
+        "QUNI": "Quercus nigra L.",
+        "QURU": "Quercus rubra L.",
+        "QUERC": "Quercus sp.",
+        "ROPS": "Robinia pseudoacacia L.",
+        "TSCA": "Tsuga canadensis (L.) Carriere",
+    }
+    metadata = {
+        "train": {
+            "url": "https://zenodo.org/record/3934932/files/IDTREES_competition_train_v2.zip?download=1",  # noqa: E501
+            "md5": "5ddfa76240b4bb6b4a7861d1d31c299c",
+            "filename": "IDTREES_competition_train_v2.zip",
+        },
+        "test": {
+            "url": "https://zenodo.org/record/3934932/files/IDTREES_competition_test_v2.zip?download=1",  # noqa: E501
+            "md5": "b108931c84a70f2a38a8234290131c9b",
+            "filename": "IDTREES_competition_test_v2.zip",
+        },
+    }
+    directories = {"train": ["train"], "test": ["task1", "task2"]}
+    image_size = (200, 200)
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        task: str = "task1",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new IDTReeS dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train" or "test"
+            task: 'task1' for detection, 'task2' for detection + classification
+                (only relevant for split='test')
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            ImportError: if laspy and pandas are are not installed
+        """
+        assert split in ["train", "test"]
+        assert task in ["task1", "task2"]
+        self.root = root
+        self.split = split
+        self.task = task
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        self.class2idx = {c: i for i, c in enumerate(self.classes)}
+        self.idx2class = {i: c for i, c in enumerate(self.classes)}
+        self.num_classes = len(self.classes)
+        self._verify()
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+        try:
+            import laspy  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "laspy is not installed and is required to use this dataset"
+            )
```

Comment:
If you move these to the respective functions where these modules are actually used, you'll only need to import once instead of twice

Reply:
There are multiple methods which use laspy and pandas so this would be just repeated code. If that's what you want then I'll make the change.

Reply:
Nah, if that's the case then leave it here